src/librustc_session/lib.rs

```diff
@@ -1,7 +1,22 @@
+#![feature(test)]
+
+// Use the test crate here so we depend on getopts through it. This allow tools to link to both
+// librustc_session and libtest.
+extern crate test as _;
+extern crate getopts;
+
 pub mod cgu_reuse_tracker;
 pub mod utils;
 #[macro_use]
 pub mod lint;
 pub mod feature_gate;
 pub mod node_id;
 pub mod parse;
+
+mod code_stats;
+pub mod config;
+pub mod filesearch;
+pub mod search_paths;
+
+mod session;
+pub use session::*;
```

Comment:
Would be better imo to explicitly re-export everything to have more control over the list.

Reply:
To be clear I expect that the `session` module becomes lib.rs in a future PR, I wanted to avoid that due to likelihood of git losing track of the rename.